modules/aruco/src/aruco.cpp

```diff
@@ -1542,22 +1542,22 @@ void _drawPlanarBoardImpl(Board *_board, Size outSize, OutputArray _img, int mar
             // remove negativity
             p1.x = p0.x - minX;
             p1.y = p0.y - minY;
-            pf.x = p1.x * float(markerZone.cols - 1) / sizeX;
-            pf.y = float(markerZone.rows - 1) - p1.y * float(markerZone.rows - 1) / sizeY;
+            pf.x = p1.x * float(markerZone.cols) / sizeX;
```

Comment:
Sorry, I dont agree.
In `getPerspectiveTransform` the maximum size is `markerZone.cols` since you have to count the length of the last pixel.

For instance, if we are in a hypothetical case where `markerZone` is 1x1 pixel, the transformation should be:
`(minX, minY) -> (0,0)`
`(maxX, maxY) -> (1,1)`

If we do what you are proposing we would get:
`(minX, minY) -> (0,0)`
`(maxX, maxY) -> (0,0)`

Which I think it is clearly wrong.